### DISCUSSION TITLE
running SPI at 80MHz

### DISCUSSION THREAD

**Discussion by jimxbob** [harbaum/galagino#50] [Q&A]
My display seems to cope with 80MHz  (#define TFT_SPICLK  80000000 )
and _I think_ it makes Galaga slightly more playable.
But here's the interesting thing - if I enable all the games, then at 80MHz both DigDug and Galaga trip the watchdog timer.
If I disable Frogger, then they work OK. Is this expected behaviour?
Reading this: https://github.com/harbaum/galagino/issues/26 I thought I'd check my ESP32 boards vn. I was at 3.0.2 which works fine for the most part at least. I downgraded to 2.0.6, but that hasn't changed the behaviour.
I can totally live with it as it is - this is such a fab project (I've built it with my 14y/o daughter), but if I can get it all working at 80MHz that would be even better.

**Comment by harbaum** [comment]
That's to be expected. The more games are installed the more CPU power is needed to distinguish between all of them during runtime. That together with the additional demands of updating the screen 60 times a second pushes it too far.

You can try to optimize the code to be less dependent on this constant game detection. But I simply never felt the urge to optimize it further ...

**Reply by jonesea** [reply]
The original hardware needs to have a memory location written to every so often otherwise the hardware WDR kicks in.  The hardware WDR needs to be kicked/cleared every 8 (?) scan lines if I remember correctly (it's the IC6A 74LS393 counter).  Wondering with the display running faster now, that the periodic software memory write to clear the WDR counter/interval is not quick enough now.

(sorry, I replied to the wrong level, I meant this to be a general comment).

**Reply by harbaum** [reply]
In order for a watchdog hardware to be triggered it actually needs to be implemented. And I simply didn't implement it. The mechanics that would trigger a reboot if the original machine simply doesn't exist in galagino.

I am quite sure that in these cases the cpu emulation is too slow it gets out of sync with the video emulation and this in turn causes a synchronization method  of freertos I use to fail which in turn triggers a reboot.

This can sure be fixed. Did you try disabling all but one arcade machine? Does it work then?
